device/core/src/internal_client.ts

```diff
@@ -105,7 +104,7 @@ export abstract class InternalClient extends EventEmitter {
         }
 
         /*Codes_SRS_NODE_INTERNAL_CLIENT_16_099: [If the transport emits a `disconnect` event while the client is subscribed to desired properties updates the retry policy shall be used to reconnect and re-enable the feature using the transport `enableTwinDesiredPropertiesUpdates` method.]*/
-        if (this._twin && this._twin.desiredPropertiesUpdatesEnabled) {
+        if (this._twin && this._twin.userRegisteredDesiredPropertiesListener) {
```

Comment:
We want to re-enable twin desired properties if the user registered a listener.
Before, we were enabling if the user registered a listener AND the transport successfully enabled twin desired properties prior.